mp/src/game/client/momentum/ui/HUD/hud_stickybombs.cpp

```diff
@@ -0,0 +1,119 @@
+// Purpose:
+//-----------------------------------------------------------------------------
+class CHudStickybombs : public CHudElement, public EditablePanel
+{
+    DECLARE_CLASS_SIMPLE(CHudStickybombs, EditablePanel);
+
+  public:
+    CHudStickybombs(const char *pElementName);
+
+    void ApplySchemeSettings(IScheme *scheme) OVERRIDE;
+    bool ShouldDraw() OVERRIDE;
+    void OnTick() OVERRIDE;
+
+  private:
+    vgui::EditablePanel *m_pStickybombsPresent;
+    vgui::EditablePanel *m_pNoStickybombsPresent;
+    vgui::Label *m_pStickyLabel;
+    vgui::Label *m_pNoStickyLabel;
+};
+
+DECLARE_HUDELEMENT(CHudStickybombs);
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+CHudStickybombs::CHudStickybombs(const char *pElementName)
+    : CHudElement(pElementName), BaseClass(NULL, "CHudStickybombs")
+{
+    Panel *pParent = g_pClientMode->GetViewport();
+    SetParent(pParent);
+
+    m_pStickybombsPresent = new EditablePanel(this, "StickybombsPresentPanel");
+    m_pNoStickybombsPresent = new EditablePanel(this, "NoStickybombsPresentPanel");
+    m_pStickyLabel = new Label(m_pStickybombsPresent, "NumStickybombsLabel", "");
+    m_pNoStickyLabel = new Label(m_pNoStickybombsPresent, "NumNoStickybombsLabel", "");
+
+    SetHiddenBits(HIDEHUD_MISCSTATUS);
+
+    LoadControlSettings("resource/ui/HudStickybombs.res");
+
+    vgui::ivgui()->AddTickSignal(GetVPanel());
+}
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+void CHudStickybombs::ApplySchemeSettings(IScheme *pScheme) { BaseClass::ApplySchemeSettings(pScheme); }
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+bool CHudStickybombs::ShouldDraw(void)
+{
+    C_MomentumPlayer *pPlayer = C_MomentumPlayer::GetLocalMomPlayer();
+
+    if (!pPlayer || !g_pGameModeSystem->GameModeIs(GAMEMODE_SJ) || !pPlayer->IsAlive())
+        return false;
+
+    if (!mom_sj_stickycount_enable.GetBool())
+        return false;
+
+    return CHudElement::ShouldDraw();
+}
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+void CHudStickybombs::OnTick(void)
+{
+    C_MomentumPlayer *pPlayer = C_MomentumPlayer::GetLocalMomPlayer();
+
+    if (!pPlayer)
+        return;
+
+    CWeaponBase *pGun = dynamic_cast<CWeaponBase *>(pPlayer->GetActiveWeapon());
+
+    if (!pGun)
+        return;
+
+    int iWeaponID = pGun->GetWeaponID();
+
+    if (iWeaponID != WEAPON_STICKYLAUNCHER)
+        return;
+
+    CMomentumStickybombLauncher *pLauncher = dynamic_cast<CMomentumStickybombLauncher *>(pGun);
```

Comment:
You may be able to save a `dynamic_cast` every frame by storing this in a private class member variable.

Initting to nullptr on constructor, and also inside `Reset()` (which gets called every player spawn) would be good, that way you can only do these casts if the member pointer is nullptr.